ndless-sdk/bin/nspire-tools

```diff
@@ -12,15 +12,13 @@ DIRNAME="$(dirname "$0")"
 
 # DIRNAME may not be in the toolchain directory if this script was symlinked.
 # try to find the symlink target	
-which readlink >/dev/null 2>&1 && readlink "$0" >/dev/null 2>&1 && {
-    SYMLINK="$(readlink "$0")"
-    
-    # On MacOS X, the readlink command hasn't the same behavior than the others operating systems based on Unix.
-    if [ "$UNAME" == "darwin" ]; then
-    	DIRNAME="$(dirname "$SYMLINK")"
-    else
-    	DIRNAME="${SYMLINK}/.."
-    fi
+{
+  (which realpath && alias realpath=realpath) || (readlink -f / && alias realpath='readlink -f') || \
```

Comment:
This is really hard to read, can it be converted into more straightforward `if foo; then bar; fi` syntax?